src/views/clusteroverview/utils/VirtctlPopup.tsx

```diff
@@ -0,0 +1,69 @@
+import * as React from 'react';
+
+import { ConsoleCLIDownloadModelGroupVersionKind } from '@kubevirt-ui/kubevirt-api/console';
+import ExternalLink from '@kubevirt-utils/components/ExternalLink/ExternalLink';
+import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import { K8sResourceCommon, useK8sWatchResource } from '@openshift-console/dynamic-plugin-sdk';
+import { Button, Popover, PopoverPosition } from '@patternfly/react-core';
+
+import { VIRTCTL_DOWNLOADS } from './constants';
+import { K8sResourceKind } from './types';
+
+import './VirtctlPopup.scss';
+
+const VirtctlPopup: React.FC = () => {
+  const { t } = useKubevirtTranslation();
+  const [cliTools] = useK8sWatchResource({
+    groupVersionKind: ConsoleCLIDownloadModelGroupVersionKind,
+    isList: true,
+    namespaced: false,
+  });
+
+  const tools = Array.isArray(cliTools) ? cliTools : [...[cliTools]];
+  // skipcq: JS-0349
+  const virtctlObj = tools?.find(
+    (tool) => (tool as K8sResourceCommon)?.metadata?.name === VIRTCTL_DOWNLOADS,
```

Comment:
u sure K8sResourceCommon and K8sResourceKind are mandatory here as hard casting?

Reply:
I typically don't hard cast unless I get errors, but I double-checked and was able to remove the hard cast to `K8sResourceCommon`, but the `K8sResourceKind` is needed to avoid an error as described in the DeepSource comment above.